Documentation/installation/k8s-install-migration.rst

```diff
@@ -0,0 +1,310 @@
+
+Overview
+========
+
+The migration process utilizes the :ref:`per-node configuration<per-node-configuration>`
+feature to selectively enable Cilium CNI. This allows for a controlled rollout
+of Cilium without disrupting existing workloads.
+
+Cilium will be installed, first, in a mode where it establishes an overlay
+but does not provide CNI networking for any pods. Then, individual nodes will
+be migrated.
+
+In summary, the process looks like:
+
+1. Install cilium in "secondary" mode.
+2. Cordon, drain, migrate, and reboot each node
+3. Remove the existing network provider
+4. (Optional) Reboot each node again
+
+
+Migration procedure
+===================
+
+Preparation
+-----------
+
+Optional: Create a :ref:`Kind <gs_kind>` cluster and install Flannel on it
+
+  .. code-block:: shell-session
+
+    $ cat <<EOF > kind-config.yaml
+    apiVersion: kind.x-k8s.io/v1alpha4
+    kind: Cluster
+    nodes:
+    - role: control-plane
+    - role: worker
+    - role: worker
+    networking:
+      disableDefaultCNI: true
+    EOF
+    $ kind create cluster --config=kind-config.yaml
+    $ kubectl apply -n kube-system --server-side -f \ |SCM_WEB|\/examples/misc/migration/install-reference-cni-plugins.yaml
+    $ kubectl apply --server-side -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml
+    $ kubectl wait --for=condition=Ready nodes --all
+
+1.  Select a **new** CIDR for pods. It must be distinct from all other CIDRs in use.
+
+    For Kind clusters, the default is ``10.244.0.0/16``. So, for this example, we will
+    use ``10.245.0.0/16``.
+
+2.  Select a **distinct** encapsulation port. For example, if the existing cluster
+    is using VXLAN, then you should either use GENEVE or configure Cilium to use VXLAN
+    with a different port.
+
+    For this example, we will use VXLAN with a non-default port of 8473. But the Cilium
+    default of GENEVE encapsulation is preferable.
+
+3.  Create a helm ``values-migration.yaml`` file based on the following example. Be sure to fill
+    in the CIDR you selected in step 1.
+
+    .. code-block:: yaml
+
+        operator:
+          unmanagedPodWatcher:
+            restart: false # Migration: Don't restart unmigrated pods
+        tunnel: vxlan # Migration: Optional: default is geneve, configure as needed
+        tunnelPort: 8473 # Migration: Optional, change only if both networks use the same port by default
+        cni:
+          customConf: true # Migration: don't install a CNI configuration file
+        ipam:
+          mode: "cluster-pool"
+          operator:
+            clusterPoolIPv4PodCIDRList: ["10.245.0.0/16"] # Migration: ensure this is distinct and unused
+        policyEnforcementMode: "never" # Migration: Disable policy enforcement
+        bpf:
+          hostLegacyRouting: true # Migration: Allow for routing between Cilium and the existing overlay
+
+4.  Optional: auto-detect settings using the :ref:`cilium-cli <install_cilium_cli>`.
```

Comment:
I don't think this should be "Optional" as if a reader decided to skip through the steps, they'd miss this step and the creation of the values-initial.yaml file which is required for the following step.

Reply:
fair enough; updated